pandas/tests/groupby/test_groupby.py

```diff
@@ -2521,6 +2521,26 @@ def test_groupby_cumprod(self):
         expected.name = 'value'
         tm.assert_series_equal(actual, expected)
 
+        # make sure that skipna works
```

Comment:
Can you split this into a new test? Make sure to add a reference to the Github issue as a comment.

Reply:
I'd find this a bit easier to read as

```python
df = pd.DataFrame({"x": [...], "gp": ['a'] * 6 + ['b'] * 6})
```

instead of `concat`.

Reply:
And maybe have a group with all-NA to ensure that is handled properly.

Reply:
I added a new test (`test_groupby_cum_skipna`) for all of the skipna-related cases. And added a groupby+cumsum/prod on all-NA values.